File: server/nodejs/src/business/domain/payments/payment-details/api-error.ts
export type APIError = {
    category?: string | null;
    code?: string | null;
    errorCode?: string | null;
    httpStatusCode?: number | null;
    id?: string | null;
    message?: string | null;
    propertyName?: string | null;
    retriable?: boolean | null;
};

File: server/python/app/application/domain/payments/payment_details/api_error.py
from dataclasses import dataclass
from typing import Optional

@dataclass
class APIError:
    category: Optional[str] = None
    code: Optional[str] = None
    error_code: Optional[str] = None
    http_status_code: Optional[int] = None
    id: Optional[str] = None
    message: Optional[str] = None
    property_name: Optional[str] = None
    retriable: Optional[bool] = None